From d92f8c0da72823885be90da59a28404088744338 Mon Sep 17 00:00:00 2001
From: Ralf Herzog <ralf.herzog@mail.com>
Date: Wed, 18 Oct 2017 16:25:36 +0000
Subject: [PATCH] LDAP bind

---
 html/includes/authentication/ldap.inc.php | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/html/includes/authentication/ldap.inc.php b/html/includes/authentication/ldap.inc.php
index e5989a1..2fac1ca 100644
--- a/html/includes/authentication/ldap.inc.php
+++ b/html/includes/authentication/ldap.inc.php
@@ -7,6 +7,7 @@ function init_auth()
     global $config, $ldap_connection;
 
     $ldap_connection = @ldap_connect($config['auth_ldap_server'], $config['auth_ldap_port']);
+    ldap_set_option($ldap_connection, LDAP_OPT_PROTOCOL_VERSION, $config['auth_ldap_version']);
 
     if ($config['auth_ldap_starttls'] && ($config['auth_ldap_starttls'] == 'optional' || $config['auth_ldap_starttls'] == 'require')) {
         $tls = ldap_start_tls($ldap_connection);
@@ -160,6 +161,8 @@ function get_userlist()
 
     $filter = '('.$config['auth_ldap_prefix'].'*)';
 
+    ldap_bind($ldap_connection, $config['auth_ldap_prefix'].$config['auth_ldap_bind_user'].$config['auth_ldap_suffix'], $config['auth_ldap_bind_password']);
+
     $search  = ldap_search($ldap_connection, trim($config['auth_ldap_suffix'], ','), $filter);
     $entries = ldap_get_entries($ldap_connection, $search);
 
-- 
2.7.4


